pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -581,7 +586,10 @@ func doNetworkInstanceSubnetSanityCheck(
 	}
 
 	// Verify Subnet doesn't overlap with other network instances
-	for _, iterStatusEntry := range ctx.networkInstanceStatusMap {
+	ctx.niStatusMapMutex.RLock()
+	statusMap := ctx.networkInstanceStatusMap
+	ctx.niStatusMapMutex.RUnlock()
```

Comment:
Probably we should unlock in defer here and inside launchHostProbe?

Reply:
@giggsoff my intention is to make a copy of the current statusMap and release the lock immediately. otherwise in the loop, it could be used by other with write access. Do you see this a problem?

Reply:
I think in particular in launchHostProbe, in that loop, it needs to publish the ni-status, which will have write to that map

Reply:
Maps are passed by reference which means that `statusMap` still points to the same memory. But as you have pointed out, we need to be careful here to avoid any deadlocks.

Reply:
Good point. let me change this to real map copy @milan-zededa